### ISSUE TITLE
Doesn't work for cloudflare

### ISSUE THREAD

**Issue by kevinvugts** [caprolactam/remix-cloudflare-sitemap#1]
Hi,

I tried to run your example in my project, but the only thing that I get when running yarn build / post build is:

```
file:///build/server/index.js:12
import { renderToReadableStream } from "react-dom/server";
         ^^^^^^^^^^^^^^^^^^^^^^
SyntaxError: Named export 'renderToReadableStream' not found. The requested module 'react-dom/server' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'react-dom/server';
const { renderToReadableStream } = pkg;

    at ModuleJob._instantiate (node:internal/modules/esm/module_job:146:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:229:5)
    at async ModuleLoader.import (node:internal/modules/esm/loader:473:24)```
    
    Can you help me out?

**Comment by caprolactam** [comment]
## TL;DR
Try this.
https://github.com/caprolactam/remix-cloudflare-sitemap/blob/73a02981c7fc244420b9564a2ad6533ffce11887/types/deps.d.ts#L1-L3
https://github.com/caprolactam/remix-cloudflare-sitemap/blob/e49db7a533224f5368391d4fdf86298b3c9e3ff3/app/entry.server.tsx#L10

## Why?
This script runs on Node.js. However, there is an issue where `renderToReadableStream` cannot be imported in a Node.js environment. I think that this is because react uses different exports for Node.js and edge environments. [[Article](This is in japanese,  but you can follow codes.)](https://zenn.dev/remew/articles/import-render-to-readable-stream-from-node)

You can work around this issue by explicitly importing `renderToReadableStream` from `react-dom/server.browser`. Additionally, since React does not provide types for it, you will need to either use `@ts-ignore` or create your own type definition file.
```
// @ts-ignore
import { renderToReadableStream } from "react-dom/server.browser";

or
// type/deps.d.ts
declare module 'react-dom/server.browser' {
  export * from 'react-dom/server'
}
```

By the way, React Router v7 supports pre-rendering. While I haven't tried it, it might be possible to use `@nasa-gcn/remix-seo` normally and add prerender option.
```
// react-router.config.ts
import type { Config } from '@react-router/dev/config'

export default {
  prerender: ['/sitemap.xml'],
} satisfies Config

```

For more details, please see [my article](https://caproweb-com.translate.goog/contents/sitemap-remix/?_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=en). Google Translate was not available due to my site settings, but it has been fixed.
